pulsar/internal/service_name_resolver.go

```diff
@@ -42,6 +42,12 @@ type pulsarServiceNameResolver struct {
 	ServiceURL   *url.URL
 	CurrentIndex int32
 	AddressList  []*url.URL
```

Comment:
Not sure why all these are public while the struct is private but that can be corrected in another MR.

Reply:
> Not sure why all these are public

I think so too.
I won't change the visibility in this PR, but we might want to do so.